crates/fuel-core/src/graphql_api/api_service.rs

```diff
@@ -220,6 +221,8 @@ where
     OnChain::LatestView: OnChainDatabase,
     OffChain::LatestView: OffChainDatabase,
 {
+    graphql_api::initialize_query_costs(config.config.costs.clone())?;
```

Comment:
a comment here about not moving this line will be nice. if it is moved later, calls to `query_costs()` will yield the default value instead of the user's configuration.

Reply:
additionally, I have a question about the graphql query complexity annotations, are they evaluated at compile time or runtime?

answered my own question - they are evaluated at runtime within a closure (expanded macro) -

![image](https://github.com/user-attachments/assets/0d949ab0-4df8-4fc9-bf84-49cf2a5bcbdf)

I assume here that for every query it will run the closure, which might be a red flag for graphql performance, because of the atomic operations done by `OnceLock::get()` in the `query_costs()` function. 

will defer to @xgreenx here, but we should load test this PR to ensure there isn't a regression.

Reply:
the load testing can happen with the image generated by this pr on devnet and testnet, let's do that first

Reply:
We don't currently have any load tests for GraphQL. Of course, we need to add them, but I think for now, it is okay. There is only one available option for implementing this change for now.

Alternative way will require unsafe code. If we want to improve performance, we can create a PR into `async-graphql` with proposed solution.

Reply:
@netrome if you can create a follow up issue for this will gladly approve

Reply:
Created https://github.com/FuelLabs/fuel-core/issues/2355